content/2019/form-decoding.md

```diff
@@ -105,29 +72,21 @@ type alias RegisterForm =
     }
 ```
 
-It's simple honesty. No interest things. It just holds user inputs as it is in `String` fields.
-
-Maybe you think that `contactType` could be enumeration type:
+It's pretty simple, not much to see.  You might be surprised that `contactType` is a string rather than an enumerated type such as:
 
 ```elm
 type ContactType
     = UseEmail
     | UsePhone
 ```
 
-But it cannot handle state of `select` tag of HTML exactly. The `value` property of `select` tag is just a string, so the `ContactType` cannot handle the possibility that the select tag has unexpected `value`.
-
-## Why Form decoding is needed?
-
-You may noticed that we need some function to convert `RegisterForm` into `Goat` because we cannot benefit from static types if using `RegisterForm` as it is even after registering. It would be much better to use dynamic typing if doing so!
+The enumerated type doesn't quite capture the semantics of how HTML deals with select tags. The `value` of a select tag is just a string, with rather weak guarantees about its possible values. Its better to store whatever HTML gives us and deal with it at the validation layer.
```

Comment:
https://www.grammarly.com/blog/its-vs-its/

This is a confusing part of English grammar, but `its` and `it's` are two different words :-\

I think `its` should be correct here.

Reply:
Thanks!
So, in this sentence, `Its` is an adjective, but I cannot understand which noun it modifies.
Could you explain more in detail, or translate this sentence into Japanese?

I understood this sentence as follows by replacing "Its" into "It's":
> __Storing whatever HTML gives us__ is better and __dealing with it at the validation layer__ is better.

Reply:
So there are actually two things going on here, relating to both words "its" and "it's".

Let's look at the first sentence:

> The `value` of a select tag is just a string, with rather weak guarantees about its possible values.

Here "its" is possessive, acting in place of a noun, referring to the string used in the first part of the sentence. I would think something like this: "its" indicates that the "possible values" belong to something. To what? To the string!

Now the second sentence:

> Its better to store whatever HTML gives us and deal with it at the validation layer.

This is actually a typo! Sorry about that. The sentence should read like this:

> It's better to store whatever HTML gives us and deal with it at the validation layer.

As cdepillabout noted, "its" and "it's" are two totally different words. Even though they look the same they are not related at all. "It's" is a contraction that expands to either "it is" or "it has". The "it is" expansion is probably more common, but they both show up. So the expanded sentence would read:

> It is better to store whatever HTML gives us and deal with it at the validation layer.

Writing it with "Its" was _incorrect_, as you noted. I wrote down a rule of thumb for it's/its but then I realized it probably helps native speakers much more than it would help a foreign speaker. Here it is anyway though:

If you're wondering whether to use "it's" with the apostrophe or "its" without, try expanding the apostrophe version to "it is" and see if the sentence makes sense. Is the "it" doing something? Then think about "its". Does the thing being referred to possess something?

Reply:
> or translate this sentence into Japanese?

I wish I could translate it for you! I'm learning Kanji, but my knowledge is nowhere near comprehensive enough to translate something. Maybe someday :)

Reply:
To tell the truth, I (and most of Japanese people) does not have a sense that "its" and "it's" are similar. In Japanese English education, we learn English not by sound but by characters. We study hard to read/write English, but not so hard to speak/listen English. It is because why Japanese people thinks "its" and "it's" are totally different. It is interesting for me that English native accidentally typo them.

It's my fault, but what I wanted to tell was just: "Its better" seems to be typo. Now I understood that cdepillabout said that it's impossible to replace **all** "its" into "it's", but what I wanted to say was that I want to replace the only "Its" of "Its better". The reason why I said like that is because I wondered: Perhaps my English knowledge would be old and it's totally okay to use "Its" instead of "It's" because I feel they are not too similar to mistake. I've seen such mistakes several times in English written by cdepillabout. (cdepillabout is American programmer living in Japan, and was working with me till last year. He speaks Japanese fluently. So I asked him to translate into English in order to understand why he thought that `its` should be OK.)

Reply:
> It is because why Japanese people thinks "its" and "it's" are totally different. It is interesting for me that English native accidentally typo them.

Now I find that *very* interesting. Mixing "it's" and "its" is one of the most common mistakes native speakers make. I wonder, do Japanese people mix up "their" and "there"? What about "then" and "than"? Those are also among the most common mistakes for native speakers.

Reply:
It's also surprising for me to hear it! We rarely mix up "their" and "there" or "then" and "than".